dbms/src/Flash/Mpp/MPPTunnel.h

```diff
@@ -270,6 +270,10 @@ class LocalTunnelSender : public TunnelSender
 
     bool push(TrackedMppDataPacketPtr && data) override
     {
+        std::lock_guard lock(mu);
```

Comment:
I am concerned that this may cause performance problems here.
Maybe we can move the `isDone` to `local_request_handler`?

Reply:
> I am concerned that this may cause performance problems here. Maybe we can move the `isDone` to `local_request_handler`?

I think that moving `isDone` to `local_request_handler` is still unsafe. As to performance, I will investigate it.